spec/system/admin/subscriptions/smoke_tests_spec.rb

```diff
@@ -242,15 +242,21 @@
         expect(page).to have_selector "#subscription-line-items .item", count: 4
 
         # Delete an existing product
-        puts "TODO: migrate to v3" # but first we need to make the actions menu more accessible.
-        Flipper.disable(:admin_style_v3) # disabling BUU for legacy products page
-        login_as_admin
-        visit spree.admin_products_path
-        within "#p_#{shop_product2.id}" do
-          accept_alert { page.find("[data-powertip=Remove]").click }
+        visit admin_products_url
+
+        product_selector = row_containing_name(shop_product2.name)
+        delete_option_selector = "a[data-controller='modal-link'].delete"
+        delete_button_selector = "input[type=button][value='Delete product']"
+        modal_selector = "div[data-modal-target=modal]"
+
+        within product_selector do
+          page.find(".vertical-ellipsis-menu").click
+          page.find(delete_option_selector).click
```

Comment:
Would this work?
```suggestion
          click_link "Delete"
```

Reply:
Thanks for the early review @dacook :pray: 

Hum.. Unfortunately not. I think first, we'd need to open the actions sub menu, which we could do with:

`page.find(".vertical-ellipsis-menu").click`

But even after that, it does not seem to work. This is the element we're trying to click I think, which is neither a button nor a link:

```
<a class="delete" data-action="click->modal-link#setModalDataSetOnConfirm click->modal-link#open" data-controller="modal-link" data-modal-link-modal-dataset-value="{&quot;data-delete-path&quot;:&quot;/admin/products_v3/11&quot;}" data-modal-link-target-value="product-delete-modal">
Delete
</a>
```

I'm not sure how we could simplify that spec line...

Reply:
Oh, fair enough. This is a classic example where the spec reveals that the markup needs improving!

I think it _should_ be a link. Hrm, but that's complicated, we can't make a useful link without JS, so it's a JS-only element. So... let's just leave this as-is, thanks for figuring out this one.